### DISCUSSION TITLE
How to change keyboard layout with Caps Lock key?

### DISCUSSION THREAD

**Discussion by optimistic5** [qtile/qtile#4402] [Q&A]
In my `autostart.sh` I have this line
`setxkbmap us,ru -option grp:caps_toggle`
which allows me to change layout with pressing Caps Lock key.
I'm not really happy with this solution  because sometimes it stopped working, so I need manually run this command again.

I want to use [KeyboardLayout](https://docs.qtile.org/en/stable/manual/ref/widgets.html?#keyboardlayout) widget.
I have this code in my `config.py`

```python
    Key([mod], "space", lazy.widget["keyboardlayout"].next_keyboard(), desc="Next keyboard layout."),
...
KeyboardLayout(
                    background=gruvbox['cyan'],
                    configured_keyboards=['us','ru']
                ),
```

It change layout with `super+space`,
**But how I can change layout with just Caps Lock?**

I tried 
`    Key(["lock"], "", lazy.widget["keyboardlayout"].next_keyboard(), desc="Next keyboard layout."),`
but it does not work.

Thank you

**Comment by tych0** [comment]
I use a custom keyboard map to accomplish something like this, which I've found to be more robust than xmodmap:

https://github.com/qtile/qtile-examples/tree/master/tych0
